src/solvers/sat/satcheck_minisat2.cpp

```diff
@@ -310,13 +287,13 @@ void satcheck_minisat_simplifiert::set_frozen(literalt a)
   if(!a.is_constant())
   {
     add_variables();
-    solver->setFrozen(a.var_no(), true);
+    static_cast<Minisat::SimpSolver*>(solver)->setFrozen(a.var_no(), true);
```

Comment:
Why is this better than the previous code?

Reply:
The only point of CRTP and declaring every single member as a template was to avoid a single cast. With virtual functions on top it is rather difficult to read. If you still don't like this raw cast could be wrapper in a method.

The intended use of CRTP is to avoid code duplication between unrelated classes while avoiding virtual functions in performance critical code. Since class was already virtual, there was no point in using CRTP here.

Reply:
FWIW I don't see how `class satcheck_minisat_no_simplifiert: public satcheck_minisat2_baset<Minisat::Solver>` ever was an instance of CRTP. There was a templated base class that could be instantiated in a number of ways. What happened was that not all instantiations would support the full interface. That seemed like a perfectly legitimate use of templates and template instantiation to me?